File: nodejs/tracing/app/sender.ts
import { OtelTracer } from "../lib/otel-tracer";
import axios from 'axios';

const tracer = new OtelTracer('nodejs-tracing-demo-sender', '0.0.0');
tracer.initialize();

async function main() {
    // Simulate some multi-step pipeline. 
    //   3 sequential jobs: job-a, job-b, job-c
    //      job-c includes 3 instances of a task, task-c, run in parallel
    const span = tracer.span('pipeline', {});
    const context = tracer.context(span);

    console.log('Starting jobs')

    for (let i = 0; i < 3; i++) {
        await axios({
            method: 'POST',
            url: 'http://localhost:3000/job',
            data: {
                b3: tracer.toB3(context),
            }
        });
    }

    console.log('All jobs finished')
    span.end();

    // Make sure all our spans are exported before exiting
    await tracer.flush();
}

main();

File: python/tracing/app/sender.py
import sys
sys.path.append('./')
import time
import requests
from lib.tracing import OtelTracer

tracer = OtelTracer(service_name='python-tracing-demo-sender', service_version='0.0.0')
tracer.initialize()


def main():
    """
    Example creating a parent and child span, and manually calling the "end" method when done.
    """
    # Create parent span, context
    span = tracer.span("pipeline")
    context = tracer.context(span)

    # Serialize context into b3
    b3 = tracer.to_b3(context)

    requests.post('http://127.0.0.1:5000/job', json = { 'b3': b3 })
    requests.post('http://127.0.0.1:5000/job', json = { 'b3': b3 })
    requests.post('http://127.0.0.1:5000/job', json = { 'b3': b3 })

    span.end()
    tracer.flush()

if __name__ == '__main__':
    main()